### DISCUSSION TITLE
Flyway veya Liquibase kullanımı

### DISCUSSION THREAD

**Discussion by halil-ulu** [gelecekbilimde/backend#45] [General]
## Tanım
Hibernate üzerinden db yönetimini Flyway veya Liquibase geçiş yapacağız. 
Şu an ve uzun vadede db enginini değiştirme düşüncemiz ve ihtiyacımız olmadığı için Flyway ile SQL yazmak daha doğru bir seçim olacaktır. @agitrubard Liquibase ise sql olarak değilde xml olarak tablo yapılarını tutar ve engine değişimlerinde kolaylık sağlıyor.

## Daha önce açılmış issuelar
https://github.com/gelecekbilimde/gelecekbilimde-backend/issues/22

### token 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	int	|
|	expired	|	boolean	|
|	revoked	|	boolean	|
|	token	|	text	|
|	token_type	|	varchar(255)	|
|	user_id	|	bigint	|
### role_permission 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	role_id	|	bigint	|
|	permission_id	|	bigint	|
### role 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	description	|	varchar(255)	|
|	is_default	|	boolean default false	|
|	role	|	varchar(55)	|
### posts_process 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	content	|	text	|
|	created_date	|	timestamp	|
|	header	|	varchar(255)	|
|	message	|	text	|
|	post_media	|	jsonb	|
|	process	|	varchar(25)	|
|	post_id	|	bigint	|
|	user_id	|	bigint	|
|	slug	|	varchar(255)	|
### posts_media 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	cover	|	boolean	|
|	created_date	|	timestamp	|
|	media_id	|	bigint	|
|	post_id	|	bigint	|
|	user_id	|	bigint	|
### post_comment 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	post_id	|	bigint	|
|	comment_id	|	bigint	|
### post 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	active	|	boolean	|
|	content	|	text	|
|	created_at	|	timestamp	|
|	header	|	varchar(255)	|
|	last_process	|	varchar(25)	|
|	like_count	|	int	|
|	user_id	|	bigint	|
|	copyright_control	|	boolean default false	|
|	dangerous_control	|	boolean default false	|
|	typo_control	|	boolean default false	|
|	slug	|	varchar(255)	|
### permission 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	description	|	varchar(255)	|
|	permission	|	varchar(255)	|
### media_group 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id		bigint	|
|	created_date		timestamp	|
|	name		varchar(50)	|
|	parent_id		int	|
|	user_id		bigint	|
### media 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	created_date	|	timestamp	|
|	media_type	|	varchar(25)	|
|	shared	|	boolean default false	|
|	title	|	varchar(100)	|
|	type	|	varchar(25)	|
|	group_id	|	bigint	|
|	user_id	|	bigint	|
### follower 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	follower_id	|	bigint	|
|	followed_id	|	bigint	|
### comment 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	comment	|	text	|
|	like_count	|	int	|
### comment 				
|	Field	|	Ext	|
|	--------------	|	--------------	|
|	id	|	bigint	|
|	description	|	varchar(255)	|
|	email	|	varchar(25)	|

**Comment by SaidMuratOzdemir** [comment]
Merhabalar,
Token'da neden expired ve revoked değerleri tutuluyor? İşlevsiz token'ları neden direkt silmiyoruz? Ayrıca token type, tokeni aldığımız yer mi demek örneğin telefonun marka-modeli?

**Reply by agitrubard** [reply]
Buradaki token eğer firebase token ise tablo ismini de ona göre belirlemek lazım, aksi hâlde uygulamadaki token ile karışabilir

**Reply by halil-ulu** [reply]
Aslında, jwt token için log tablosu cihetinde yapmıştım.
Kullanıcı login sayısı ne sıklıkla token talebi oluyor gibi.
Token type bearer token yazıyor şuanlık ama google authentication gibi kurgularda tokenı nerden aldığını tutmak için gibi gibi..
değerlendirelim olamadı tabloyu uçururuz

**Reply by agitrubard** [reply]
Bunun yerine Audit Log gibi yapılar var bununla her kullanıcının yaptığı aksiyonları tutabiliyorsunuz

**Reply by halil-ulu** [reply]
evet evet o topa girmeden hızlıca db ye kayıt ederim diye düşündüm. Autdit logları koda fazla bulaştırmadan yapmak lazım. O yüzden 2. 3. fazlarda yapararız diye düşünmüştük :)

**Comment by halil-ulu** [comment]
Konunun ortaya çıkmasıyla ilintili olarak ilk 
Tablo ismine benzer alanların kendini amacını anlatmıyor olmasıydı. Bu alanları değiştireceğim 
@ferhatozcelik buraya ticket için eklediğin tablolarıda yazarmısın
